lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -238,11 +446,32 @@ private static function get_block_nodes( $theme_json, $selectors = array() ) {
 	 * @return string Styles for the block.
 	 */
 	public function get_styles_for_block( $block_metadata ) {
-		$node         = _wp_array_get( $this->theme_json, $block_metadata['path'], array() );
-		$selector     = $block_metadata['selector'];
-		$settings     = _wp_array_get( $this->theme_json, array( 'settings' ) );
-		$declarations = static::compute_style_properties( $node, $settings );
-		$block_rules  = '';
+
+		$node = _wp_array_get( $this->theme_json, $block_metadata['path'], array() );
+
+		$selector = $block_metadata['selector'];
+		$settings = _wp_array_get( $this->theme_json, array( 'settings' ) );
+
+		// Attempt to parse a pseudo selector (e.g. ":hover") from the $selector ("a:hover").
+		$pseudo_matches = array();
+		preg_match( '/:[a-z]+/', $selector, $pseudo_matches );
+		$pseudo_selector = $pseudo_matches[0] ?? null;
+
+		// Get a reference to element name from path.
+		// $block_metadata['path'] = array('styles','elements','link');
+		$current_element = $block_metadata['path'][ count( $block_metadata['path'] ) - 1 ];
```

Comment:
I added an `echo` here like this:

```php
echo '['.implode("][", $block_metadata['path'])."]\n";
```

And these are all the paths it printed:

```
[styles][blocks][core/audio]
[styles][blocks][core/button]
[styles][blocks][core/table]
[styles][blocks][core/video]
[styles][blocks][core/embed]
[styles][blocks][core/image]
[styles][blocks][core/code]
[styles][blocks][core/gallery]
[styles][blocks][core/list]
[styles][blocks][core/navigation]
[styles][blocks][core/post-title]
[styles][blocks][core/post-date]
[styles][blocks][core/pullquote]
[styles][blocks][core/search]
[styles][blocks][core/separator]
[styles][blocks][core/quote]
[styles][blocks][core/site-tagline]
[styles][blocks][core/site-title]
[styles]
[styles][elements][h1]
[styles][elements][h2]
[styles][elements][h3]
[styles][elements][h4]
[styles][elements][h5]
[styles][elements][h6]
[styles][elements][link]
[styles][elements][link]
```

I'm a bit worried about the element-specific logic reasoning about the blocks and other keys as well. This looks safer to me:

```suggestion
		// Make sure that $block_metadata['path'] is ['styles', 'element', $current_element] and not e.g. just ['styles']
		$is_processing_element = count( $block_metadata['path'] ) === 3 && $block_metadata['path'][1] === 'elements';
		$current_element = $is_processing_element ? $block_metadata['path'][ count( $block_metadata['path'] ) - 1 ] : nulll;
```

What do you think?

Reply:
Yes that is a good point. This logic is somewhere else in the class as well I think 🤔

Reply:
@adamziel I pushed a change which fixes the tests. Does that work for you?

Reply:
Good catch and your fix looks great!